pkg/kapp/resourcesmisc/custom_waiting_resource.go

```diff
@@ -96,6 +98,12 @@ func (s CustomWaitingResource) IsDoneApplying() DoneApplyState {
 					continue
 				}
 				if condMatcher.Success {
+					if condMatcher.SupportsUnblockingChanges {
```

Comment:
location of this impl seems to imply that it only works with custom waiting resource [0] which potentially becomes problematic if you want to "unblock" x but still want regular waiting logic. any thoughts?

[0] https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/clusterapply/converged_resource_factory.go#L47

Reply:
So in the e2e test example, we should not be forced to add the `Available` condition for the deployment as that's something we already have for deployments, right?
Hmm..., adding that condition did bother me but I didn't worry much about it thinking that it would mostly be used for custom resources. But this is definitely a valid concern, I will think more about it.

Reply:
> So in the e2e test example, we should not be forced to add the Available condition for the deployment as that's something we already have for deployments, right?

as soon as you use custom waiting, code wont use "default" waiting rules (like for Deployment)

Reply:
Yep, and I am not able to think of a simple way to make this work for all resources (not just custom waiting resources) using `waitRules` because of this. 
Maybe we want to introduce some sort of `unblockingRules` that are only used for unblocking blocked changes when one of the mentioned conditions matches? (Although this sounds a bit redundant from user perspective, so maybe not the best way to go about it)

Reply:
We discussed to go forward with this approach for now and work on updating the logic for custom waiting resources so that they can be used along with default waiting rules available for other resources. I will create an issue for this and resolve this thread.

Reply:
Created a new [issue](https://github.com/vmware-tanzu/carvel-kapp/issues/512) to track this enhancement.